### PR TITLE
first PR - getting things started

### PR DESCRIPTION
**Pull request by chenshay**

### CODE REVIEW

**Comment by webhamm** [comment]
Interesting.. Wonder what this will be.

File: amp-viewer.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class AmpViewer {
```
Comment (dvoytenko) [inline amp-viewer.js]: License + jsdoc. Also this should go into `src/`

File: amp-viewer.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ 
+ /**
+  * @param {!Element} hostEl the element to attatch the iframe to.
```
Comment (dvoytenko) [inline amp-viewer.js]: `hostElement`

File: amp-viewer.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  * @param {string} ampUrl the AMP page url.
+  */
+ constructor(hostEl, ampUrl) {
+  /** @type {!Element} */
```
Comment (dvoytenko) [inline amp-viewer.js]: `@private`

File: amp-viewer.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  this.hostEl_ = hostEl;
+
+  /** @type {string} */
```
Comment (dvoytenko) [inline amp-viewer.js]: `@private`

File: amp-viewer.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  this.ampUrl_ = ampUrl;
+
+  this.attach();
```
Comment (dvoytenko) [inline amp-viewer.js]: Generally, given multitude of parameters and customizations, I cannot imagine we can attach iframe right away.

File: amp-viewer.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+ }
+ 
+ attach() {
```
Comment (dvoytenko) [inline amp-viewer.js]: jsdoc

File: amp-viewer.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+   let iframe = document.createElement('iframe');
```
Comment (dvoytenko) [inline amp-viewer.js]: This should be a private class var.

File: amp-viewer.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+   this.hostEl_.appendChild(iframe);
+   iframe.src = this.ampUrl_;
+ }
+
+}
```
Comment (dvoytenko) [inline amp-viewer.js]: nit: terminating new line

File: example/platform.html [hunk 9]
```diff
@@ -0,0 +1,105 @@
+<!doctype html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>Viewer</title>
+  <meta name="viewport" content="width=device-width,initial-scale=1,minimum-scale=1,maximum-scale=1,user-scalable=no,minimal-ui">
+  <link href='https://fonts.googleapis.com/css?family=Questrial' rel='stylesheet' type='text/css'>
+  <script src="../amp-viewer.js"></script>
```
Comment (dvoytenko) [inline example/platform.html]: Probably ok for first commit, but give unpackaged/ES6 nuances won't work very soon. You'll need to have a sample binary built.

File: amp-viewer.js (hunk 3 above)
Comment (chenshay) [inline amp-viewer.js]: done

File: amp-viewer.js (hunk 4 above)
Comment (chenshay) [inline amp-viewer.js]: done

File: amp-viewer.js (hunk 1 above)
Comment (chenshay) [inline amp-viewer.js]: done

File: amp-viewer.js (hunk 2 above)
Comment (chenshay) [inline amp-viewer.js]: done

File: amp-viewer.js (hunk 5 above)
Comment (chenshay) [inline amp-viewer.js]: done

File: amp-viewer.js (hunk 7 above)
Comment (chenshay) [inline amp-viewer.js]: done

File: amp-viewer.js (hunk 8 above)
Comment (chenshay) [inline amp-viewer.js]: done

File: example/platform.html (hunk 9 above)
Comment (chenshay) [inline example/platform.html]: acknowledged

File: src/amp-viewer.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * @fileoverview This file is a Viewer for AMP Documents.
```
Comment (dvoytenko) [inline src/amp-viewer.js]: No need for `@fileoverview` here.

File: src/amp-viewer.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+  * @param {string} ampUrl the AMP page url.
+  */
+ constructor(hostElement, ampUrl) {
+  /** @private {!Element} */
+  this.hostElement_ = hostElement;
+
+  /** @private {string} */
+  this.ampUrl_ = ampUrl;
+
+  /** @private {Element} */
```
Comment (dvoytenko) [inline src/amp-viewer.js]: `?Element`. Basically, the rule is: if it's nullable - always use "?"

Comment (chenshay) [inline src/amp-viewer.js]: done

File: src/amp-viewer.js (hunk 10 above)
Comment (chenshay) [inline src/amp-viewer.js]: done

**Pull request merged**